restfull-api/controllers/todo.js

```diff
@@ -49,7 +49,11 @@ class TodoController {
     if (!todo) {
       return res.status(204).end();
     }
-    await todo.deleteOne();
+    if (todo) {
```

Comment:
No need for this check, you have already handled the `!todo` path in the above if.